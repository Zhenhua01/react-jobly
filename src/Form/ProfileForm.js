import { useState, useEffect, useContext } from "react";
import userContext from '../Context/userContext';
import { useNavigate } from "react-router-dom";
import JoblyApi from '../utils/api';

const INITIAL_FORM_DATA = {
  username: '',
  firstName: '',
  lastName: '',
  email: ''
};

/** Form for displaying user profile.
 *
 * Props:
 * - update: function to call in parent.
 *
 * State:
 * - formData
 *
 * RoutesList -> ProfileForm ->
 */

function ProfileForm({ update }) {
  const navigate = useNavigate();
  const { user } = useContext(userContext);
  console.log('user context', user)
  const [formData, setFormData] = useState(INITIAL_FORM_DATA);

  /** Update form input. */
  function handleChange(evt) {
    const { name, value } = evt.target;

    setFormData(formData => ({
      ...formData,
      [name]: value,
    }));
  }

  /** Call parent function and clear form. */
  async function handleSubmit(evt) {
    evt.preventDefault();
    try {
      await update(formData);
      navigate("/companies");
    }
    catch {

    }
  }

  return (
    <form className="ProfileForm" onSubmit={handleSubmit}>

      <div className="mb-3">
        <label htmlFor="username">Username</label>
        <input
          id="username"
          name="username"
          className="form-control"
          onChange={handleChange}
          value={formData.username}
          aria-label="username"
          required
          disabled
        />
      </div>

      <div className="mb-3">
        <label htmlFor="firstName">First Name</label>
        <input
          id="firstName"
          name="firstName"
          className="form-control"
          onChange={handleChange}
          value={formData.firstName}
          aria-label="firstName"
          required
        />
      </div>

      <div className="mb-3">
        <label htmlFor="lastName">Last Name</label>
        <input
          id="lastName"
          name="lastName"
          className="form-control"
          onChange={handleChange}
          value={formData.lastName}
          aria-label="lastName"
          required
        />
      </div>

      <div className="mb-3">
        <label htmlFor="email">Email</label>
        <input
          id="email"
          name="email"
          className="form-control"
          onChange={handleChange}
          value={formData.email}
          aria-label="email"
          required
        />
      </div>

      <div className="mb-3">
        <button className="btn-primary btn ProfileForm-addBtn">
          Submit
        </button>
      </div>

    </form>
  );
}

export default ProfileForm;
