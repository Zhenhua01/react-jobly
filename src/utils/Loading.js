import React from "react";
import "./Loading.css";


/** Loading message to display when making API calls for data. */

function Loading() {
  return (
    <div className="Loading">
      Loading ...
    </div>
  );
}

export default Loading;