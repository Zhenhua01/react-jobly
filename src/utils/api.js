import axios from "axios";

// Use port 3001 when connecting to local backend server
// const BASE_URL = process.env.REACT_APP_BASE_URL || "http://localhost:3001";

// Use heroku deployed backend when local server not available
const BASE_URL =
    process.env.REACT_APP_BASE_URL || "https://jobly-zhl-exp.herokuapp.com";

/** API Class.
 *
 * Static class tying together methods used to get/send to to the API.
 * There shouldn't be any frontend-specific stuff here, and there shouldn't
 * be any API-aware stuff elsewhere in the frontend.
 */

 class JoblyApi {

  static token = "";

  static async request(endpoint, data = {}, method = "get") {
    console.debug("API Call:", endpoint, data, method);

    const url = `${BASE_URL}/${endpoint}`;
    const headers = { Authorization: `Bearer ${JoblyApi.token}` };
    const params = (method === "get")
        ? data
        : {};

    try {
      return (await axios({ url, method, data, params, headers })).data;
    } catch (err) {
      console.error("API Error:", err.response);
      let message = err.response.data.error.message;
      throw Array.isArray(message) ? message : [message];
    }
  }

  // Individual API routes

  /** Get details on a company by handle. */

  static async getCompany(handle) {
    let res = await this.request(`companies/${handle}`);
    return res.company;
  }

  /** Get all companies or companies matching search term. */

  static async getCompanies(data = {}) {
    let res = await this.request(`companies`, data);
    return res.companies;
  }

  /** Get all jobs or jobs matching search term. */

  static async getJobs(data = {}) {
    let res = await this.request(`jobs`, data);
    return res.jobs;
  }

  /** Registers a new user. */

  static async signup(data) {
    let res = await this.request(`auth/register`, data, "post");
    return res.token;
  }

  /** Logins user. */

  static async login(data) {
    let res = await this.request(`auth/token`, data, "post");
    return res.token;
  }

  /** Update user information. */

  static async update(username, data) {
    let res = await this.request(`users/${username}`, data, "patch");
    return res.user;
  }

  /**Get user information. */

  static async getUserInfo(username) {
    let res = await this.request(`users/${username}`);
    return res.user;
  }

  /**Add job to user applications. */

  static async applyToJob(username, id) {
    await this.request(`users/${username}/jobs/${id}`, {}, "post");
    let res = await this.request(`users/${username}`);
    return res.user;
  }
}


export default JoblyApi;