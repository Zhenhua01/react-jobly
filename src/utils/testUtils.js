import React from "react";
import userContext from "../context/userContext";

const demoUser = {
  username: "testuser",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
  photo_url: null,
  applications: [],
};

const UserProvider = ({
  children,
  user = demoUser,
  hasAppliedToJob = () => false,
}) => (
  <userContext.Provider value={{ user, hasAppliedToJob }}>
    {children}
  </userContext.Provider>
);

export { UserProvider };
