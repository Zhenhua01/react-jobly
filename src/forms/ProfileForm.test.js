import React from "react";
import { render } from "@testing-library/react";
import ProfileForm from "./ProfileForm";
import { MemoryRouter } from "react-router";
import { UserProvider } from "../utils/testUtils";


it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <ProfileForm />
      </UserProvider>
    </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});


