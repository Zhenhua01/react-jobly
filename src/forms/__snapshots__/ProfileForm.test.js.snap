// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="ProfileForm"
  >
    <div
      class="mb-3"
    >
      <label
        for="username"
      >
        Username
      </label>
      <input
        aria-label="username"
        class="form-control"
        disabled=""
        id="username"
        name="username"
        required=""
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        for="firstName"
      >
        First Name
      </label>
      <input
        aria-label="firstName"
        class="form-control"
        id="firstName"
        name="firstName"
        required=""
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        for="lastName"
      >
        Last Name
      </label>
      <input
        aria-label="lastName"
        class="form-control"
        id="lastName"
        name="lastName"
        required=""
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        for="email"
      >
        Email
      </label>
      <input
        aria-label="email"
        class="form-control"
        id="email"
        name="email"
        required=""
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <button
        class="btn-primary btn ProfileForm-addBtn"
      >
        Submit
      </button>
    </div>
  </form>
</DocumentFragment>
`;
